src/main/node/node.ts

```diff
@@ -0,0 +1,124 @@
+import { exec } from 'child_process';
+import { type } from 'os';
+import { appendLog } from '../logger';
+
+class Platform {
+  public os: string;
+
+  constructor(os: string) {
+    this.os = os;
+  }
+
+  public static WINDOWS: Platform = new Platform('Windows_NT');
+
+  public static LINUX: Platform = new Platform('Linux');
+
+  public static MAC: Platform = new Platform('Darwin');
+
+  private static supportedPlatforms: Platform[] = [
+    Platform.WINDOWS,
+    Platform.LINUX,
+    Platform.MAC,
+  ];
+
+  public static Current(): Platform {
+    return new Platform(type());
+  }
+
+  public IsSupported(): boolean {
+    return Platform.supportedPlatforms.some(({ os }) => os === this.os);
+  }
+
+  public InstallNodeManager(): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      const managerInstallCommand = this.NodeManagerInstallationCommand();
+      if (managerInstallCommand !== undefined) {
+        exec(managerInstallCommand as string, (error) => {
+          if (error) {
+            appendLog(`Node.js Manager installation failed.`);
+            reject(error);
+          }
+          appendLog(`Finished installation of node manager.`);
+          resolve();
+        });
+      }
+      const message = `Node manager installation command unspecified.`;
+      reject(message);
+    });
+  }
+
+  public InstallNodeJs(): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      const nodeInstallCommand = this.NodeInstallationCommand();
+      if (nodeInstallCommand !== undefined) {
+        exec(nodeInstallCommand as string, (error) => {
+          if (error) {
+            appendLog(`Node installation failed.`);
+            reject(error);
+          }
+          appendLog(`Finished installation of node.`);
+          resolve();
+        });
+      }
+      const message = `Node installation command unspecified.`;
+      reject(message);
+    });
+  }
+
+  private NodeManagerInstallationCommand(): string | undefined {
+    if (this === Platform.WINDOWS) return 'winget install nvs';
+    if (this === Platform.LINUX || this === Platform.MAC)
+      return 'npm install -g n';
```

Comment:
@kmieciakm so we're not moving over to installation form dist in the end?

Reply:
Dodane